Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -117,35 +174,94 @@ private class CustomCell: UITableViewCell {
+            
+            // Determine rotation / scaling amounts by the distance to the edge
+            var rotation = (distanceFromCenter / halfWidth) * totalRotationInRadians
+            rotation *= self.originalCenter.x - self.backgroundHolder.center.x > 0 ? -1 : 1
+            var scale = 1 - (distanceFromCenter / halfWidth) * (1 - TabCellUX.totalScale)
+            let alpha = 1 - (distanceFromCenter / halfWidth) * (1 - TabCellUX.totalAlpha)
+            
+            let rotationTransform = CGAffineTransformMakeRotation(rotation)
+            let scaleTransform = CGAffineTransformMakeScale(scale, scale)
+            let combinedTransform = CGAffineTransformConcat(rotationTransform, scaleTransform)
+            
+            self.backgroundHolder.transform = combinedTransform
+            self.backgroundHolder.alpha = alpha
+            break
+            
+        case .Cancelled:
+            self.backgroundHolder.center = self.originalCenter
+            self.backgroundHolder.transform = CGAffineTransformIdentity
+            self.backgroundHolder.alpha = 1
+            break
+            
+        case .Ended:
+            if (abs((Int)(self.backgroundHolder.center.x - self.center.x)) > TabCellUX.deleteThreshold) {
+                let velocity = recognizer.velocityInView(self)
+                let actualVelocity = max(Double(abs(velocity.x)), TabCellUX.minExitVelocity)
+                
+                // Calculate the edge to calculate distance from
+                let edgeX = velocity.x > 0 ? CGRectGetMaxX(self.frame) : CGRectGetMinX(self.frame)
+                var distance
+                    = CGFloat((self.backgroundHolder.frame.size.width / 2) + abs(self.backgroundHolder.center.x - edgeX))
+                
+                // Determine which way we need to travel
+                distance *= velocity.x > 0 ? 1 : -1
+                
+                let timeStep: NSTimeInterval = Double(abs(distance)) / actualVelocity
+                UIView.animateWithDuration(timeStep, animations: {
+                    let animatedPosition
+                        = CGPoint(x: self.backgroundHolder.center.x + distance, y: self.backgroundHolder.center.y)
+                    self.backgroundHolder.center = animatedPosition
+                }, completion: { finished in
+                    if finished {
+                        self.backgroundHolder.hidden = true
+                        self.tabDelegate?.tabCellDidSwipeToDelete(self)
+                    }
+                })
+            } else {
+                UIView.animateWithDuration(TabCellUX.recenterAnimationDuration, animations: {
+                    self.backgroundHolder.transform = CGAffineTransformIdentity
+                    self.backgroundHolder.center = self.originalCenter
+                    self.backgroundHolder.alpha = 1
+                })
+            }
+            break
```

Comment:
Is it possible to do as much as possible using the `CGFloat` type in the above code for `.Changed` and `.Ended` ? There is a lot of casting between `CGFloat`, `Double` and `Int` going on and I think most of that can go away.